catalogue/catalogue.py

```diff
@@ -58,28 +58,20 @@ def modified_walk(folder, ignore_subdirs=[], ignore_exts=[], ignore_dot_files=Tr
     list[str]
         A list of accepted paths
     '''
+    assert os.path.exists(folder), "Path {} does not exist".format(folder)
+
     path_list = []
     for path, directories, files in os.walk(folder):
         # loop over files in the top directory
         for f in sorted(files):
-            root, ext = os.path.splitext(f)#[1]
+            root, ext = os.path.splitext(f)
             if not (
-                (ext in ignore_exts) or (
-                ignore_dot_files and root.startswith("."))):
-                # path_list.append(os.path.join(*directories, f))
+                (ext in ignore_exts) or
+                (ignore_dot_files and root.startswith(".")) or
+                (path in ignore_subdirs)
+                ):
                 path_list.append(os.path.join(path, f))
-        for s in sorted(directories):
```

Comment:
I think your code looks correct to me based on what `os.walk` does. I'm a bit confused because the inclusion of the second loop in the previous version of the code seemed like it might be "incorrect" in that it hashes the file once per directory level as it recurses (though I guess if it produces consistent hashes it would be "correct" from a practical standpoint). So I just wanted to flag this and ensure that this is the behavior that we want.

Reply:
Yes the previous version was incorrect for the reasons you describe. You'd hash the same file multiple times (once per directory level). You'd get a consistent hash but it is not the hash/behaviour you would expect. This was the only part of the original code that I never  really checked as it seemed to work fine, which is why I missed this before.